x-pack/plugins/apm/server/plugin.ts

```diff
@@ -120,16 +128,25 @@ export class APMPlugin implements Plugin<APMPluginSetup> {
         elasticCloud: createElasticCloudInstructions(plugins.cloud),
       };
     });
+
     plugins.features.registerFeature(APM_FEATURE);
+    plugins.licensing.featureUsage.register(
+      APM_SERVICE_MAPS_FEATURE_NAME,
+      APM_SERVICE_MAPS_LICENSE_TYPE
+    );
 
-    createApmApi().init(core, {
-      config$: mergedConfig$,
-      logger: this.logger!,
-      plugins: {
-        observability: plugins.observability,
-        security: plugins.security,
-        ml: plugins.ml,
-      },
+    core.getStartServices().then(([_coreStart, pluginsStart]) => {
+      createApmApi().init(core, {
+        config$: mergedConfig$,
+        logger: this.logger!,
+        plugins: {
+          licensing: (pluginsStart as { licensing: LicensingPluginStart })
```

Comment:
it would be nice if we could have a mapped type for plugin setup and start contractions.